spec/features/results/response_form_tree_handling_spec.rb

```diff
@@ -94,8 +100,92 @@
     end
   end
 
+  describe "form submission" do
+    scenario "submitting response" do
+      visit new_hierarchical_response_path(params)
+
+      select2(user.name, from: "response_user_id")
+      fill_in("response_root_children_0_children_0_value", with: "1")
+      drop_in_dropzone(Rails.root.join("spec", "fixtures", "media", "audio", "powerup.mp3"))
+      expect(page).to have_content("The uploaded file was not an accepted format.")
+      drop_in_dropzone(Rails.root.join("spec", "fixtures", "media", "images", "the_swing.jpg"))
+      select("Animal")
+      select("Dog")
+      fill_in("response_root_children_3_children_0_children_0_children_0_value", with: "4561")
+      fill_in("response_root_children_3_children_0_children_1_children_0_children_0_value", with: "7891")
+      all("a.add-repeat").last.click
+      fill_in("response_root_children_3_children_1_children_0_children_0_value", with: "4562")
+      fill_in("response_root_children_3_children_1_children_1_children_0_children_0_value", with: "7892")
+      click_button("Save")
+
+      expect(page).to have_content("Response is invalid")
+      expect_value("#response_root_children_3_children_0_children_0_children_0_value", "4561")
+      expect_value("#response_root_children_3_children_0_children_1_children_0_children_0_value", "7891")
+      expect_value("#response_root_children_3_children_1_children_0_children_0_value", "4562")
+      expect_value("#response_root_children_3_children_1_children_1_children_0_children_0_value", "7892")
+
+      fill_in("response_root_children_0_children_0_value", with: "123")
+      click_button("Save")
+
+      expect(page).to_not have_content("Response is invalid")
+
+      response = Response.last
+      visit hierarchical_response_path(params.merge(id: response.shortcode))
```

Comment:
Don't need to worry about testing `#show` in this spec. Let's keep it to the form. Show comes in a later issue.